torchdata/dataloader2/reading_service.py

```
@@ -537,12 +547,14 @@ def finalize_iteration(self) -> None:
 
     # Sequential Order
     def checkpoint(self) -> bytes:
-        states = []
-        for rs in self.reading_services:
-            states.append(rs.checkpoint())
+        # states = []
+        # for rs in self.reading_services:
+        #     states.append(rs.checkpoint())
+        # return states
+        raise NotImplementedError
```

The implementation doesn't do anything (yet). Let's raise an exception instead?